### ISSUE TITLE
EventHandler on React Component without implementing it from the component side

### ISSUE THREAD

**Issue by erku123** [facebook/react#9898]
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
To add event handler to components outer layer, it has to be implemented from inside the component as well, like: `return <div onClick={this.props.clickHandler}></div>`  without that can't add event handlers from parent. While something like this should work in my opinion: `<SomeComponent onClick={() => {/*do the stuff*/}} />`

example:

```
export default class SomeComponent extends React.Component {
...
   render() {

   /*Doesn't have any event handler logic*/
   return <div>...content</div>
  }
};

....
/*Isn't working. But in my opinion it should/could as well?*/
render() {
   return(<div><SomeComponent  onClick={() => {console.log('the click worked!')}/></div>
};

/*Is working.  Yet I think the last way would be cleaner.
render() {
   return(<div onClick={() => {console.log('the click worked')}><SomeComponent /> </div>
}

```

So to sum it up: what I request is that event handler props added to Component would pass on to the outer layer of the Component.

**Comment by bvaughn** [comment]
The behavior you've described could be convenient in some cases (like the one you mention) but I don't think it's a good generic solution.

Firstly it would require React renderers (like DOM) to maintain an explicit whitelist of event handlers and treat them differently than all other component props. In general, whitelists are something we're moving away from (eg DOM attributes) as they bloat the codebase and can be hard to maintain.

Beyond just a whitelist, DOM would presumably also need to map which HTML elements dispatched which events to know that an `onClick` should be passed to a `div` but eg `onChange` should not. Otherwise what should happen in ambiguous cases like this?
```jsx
class SomeComponent extends React.Component {
  render() {
    // If your component assumes that React will pass-thru onClick to <div>,
    // What should it do with onChange? Presumably that's intended for <select>?
    return (
      <div>
        <select>...</select>
      </div>
    );
  }
};

// example
<SomeComponent  onClick={...} onChange={...}/>
```

I think this proposal also breaks encapsulation of components. Users of a component shouldn't necessarily know/care-about the DOM elements it composes- only that it has a specific external API. Maybe your component wants to intercept events, run some custom logic, and then allow them to propagate (or not).

The team has been considering ways to improve event subscription/handling in future versions of React. (That's a bit orthogonal to this request but I mention it in passing just to note that this is an area that may be changing in the future as well.)

Thanks for the proposal and the the examples! I'm going to close it for now though as I don't think it's something we can/should accommodate. 😄  Cheers!
